### PR TITLE
Added filter by metadata for warehouse query

### LINKED ISSUE #13345: Filter warehouses by metadata
There are stores that have hundreds of warehouses, for performing updates metadata can be useful as a lookup mechanism instead of using externalReference

### PR DESCRIPTION
**Pull request by Smit-Parmar**
I want to merge this change because
it closes #13345 

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -93,6 +93,7 @@ Shipping methods can be removed by the user after it has been assigned to a chec
   - Add `CheckoutLine.problems` field
   - Add `CheckoutSettingsInput` to `ChannelCreateInput` & `ChannelUpdateInput`
   - Add `checkoutSettings` field to `Channel`
+- Add Filter warehouses by metadata -#13345 by @Smit-Parmar
```
Comment (8r2y5) [inline CHANGELOG.md:96]: Missing space

Comment (Smit-Parmar) [reply CHANGELOG.md:96]: @8r2y5 Above lines are sub points. You can verify that by opening full file.

Comment (Smit-Parmar) [reply CHANGELOG.md]: Done!

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by 8r2y5 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -93,6 +93,7 @@ Shipping methods can be removed by the user after it has been assigned to a chec
   - Add `CheckoutLine.problems` field
   - Add `CheckoutSettingsInput` to `ChannelCreateInput` & `ChannelUpdateInput`
   - Add `checkoutSettings` field to `Channel`
+- Add Filter warehouses by metadata - #13345 by @Smit-Parmar
```
Comment (maarcingebala) [inline CHANGELOG.md:96]: Can you move this entry to the 3.16 section? 3.15 scope is already closed.

**Review by szczecha — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Smit-Parmar) [reply CHANGELOG.md]: @maarcingebala Moved

**Pull request merged**